File: src/client/test/app/PushButtonTest.js
/**
 * A test class for the echopoint.PushButton client-side component.
 * Displays a sample button and applies standard styles.
 *
 * @author Rakesh 2009-02-24
 * @version $Id: PushButtonTest.js,v 1.1.1.1 2010-04-01 09:47:33 perxi Exp $
 */
echopoint.test.PushButtonTest = Core.extend(
{
  button: null,
  label: null,
  select: null,
  count: 0,

  $construct: function( testArea )
  {
    testArea.add( this._createPushButton() );
    testArea.add( this._createEnabled() );
    testArea.add( this._createLabel() );
  },

  _createPushButton: function()
  {
    this.button = new echopoint.PushButton(
    {
      renderId: "echopointUnitTestPushButton",
      styleName: "Default",
      events:
      {
        action: Core.method( this, this._actionPerformed )
      }
    } );

    return this.button;
  },

  _createEnabled: function()
  {
    this.select = new Echo.SelectField(
    {
      renderId: "echopointUnitTestPushButtonSelectField",
      style: "Default",
      items: [ "Enabled", "Disabled"],
      events:
      {
        action: Core.method( this, this._enabled )
      }
    });

    return this.select;
  },

  _createLabel: function()
  {
    this.label = new Echo.Label(
    {
      renderId: "echopointUnitTestPushButtonLabel",
      style: "Default",
      text: "Action Label"
    } );

    return this.label;
  },

  _actionPerformed: function()
  {
    this.label.set( "text", "Button clicked: " + ++this.count );
  },

  _enabled: function()
  {
    if ( 1 == this.select.get( "selection" ) )
    {
      this.button.setEnabled( false );
    }
    else
    {
      this.button.setEnabled( true );
    }
  }
});

File: src/server-java/test/echopoint/PushButtonTest.java
package echopoint;

import static org.junit.Assert.assertEquals;
import nextapp.echo.app.Component;
import nextapp.echo.app.Extent;
import nextapp.echo.app.Label;
import nextapp.echo.app.SelectField;
import nextapp.echo.app.event.ActionEvent;
import nextapp.echo.app.event.ActionListener;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * Unit test for the {@link echopoint.PushButton} component.
 *
 * @author Rakesh Vidyadharan 2009-02-24
 * @version $Id: PushButtonTest.java,v 1.1.1.1 2010-04-01 09:47:30 perxi Exp $
 */
public class PushButtonTest extends AbstractTest<PushButton>
{
  static int count;
  static final String command = "actionCommand";

  @BeforeClass
  public static void init()
  {
    count = 0;
    set( new PushButton() );
  }

  @Test
  public void renderId()
  {
    final String renderId = "echopointUnitTestPushButton";
    getComponent().setRenderId( renderId );
    assertEquals( "Ensuring renderId set", renderId,
        getComponent().getRenderId() );
  }

  @Test
  public void actionCommand()
  {
    getComponent().setActionCommand( command );
    assertEquals( "Ensuring action command set", command,
        getComponent().getActionCommand() );
  }

  @Test
  public void width()
  {
    final int width = 500;
    getComponent().setWidth( new Extent( width ) );
    assertEquals( "Ensuring that width is set", width,
        getComponent().getWidth().getValue() );
  }

  @Test
  public void height()
  {
    final int height = 300;
    getComponent().setHeight( new Extent( height ) );
    assertEquals( "Ensuring that height is set", height,
        getComponent().getHeight().getValue() );
  }

  @Test
  public void text()
  {
    final String text = "Click Me!";
    getComponent().setText( text );
    assertEquals( "Ensuring that text is set", text, getComponent().getText() );
  }

  @Test
  public void toolTip()
  {
    final String text = "Click to test!";
    getComponent().setToolTipText( text );
    assertEquals( "Ensuring that tool tip text is set", text,
        getComponent().getToolTipText() );
  }

  @AfterClass
  public static void finish()
  {
    final Component content = Application.getContent().getTestArea();
    content.removeAll();
    final Label label = new Label( "Action command area" );

    ( (PushButton) get() ).addActionListener( new ActionListener() {
      private static final long serialVersionUID = 1l;

      public void actionPerformed( final ActionEvent event )
      {
        label.setText( "Button clicked: " + ++count );
        assertEquals( "Ensuring that action command is set",
            command, event.getActionCommand() );
      }
    });

    content.add( get() );
    content.add( createSelectField() );
    content.add( label );
  }

  private static SelectField createSelectField()
  {
    final String[] values = { "Enabled", "Disabled" };
    final SelectField select = new SelectField( values );
    select.addActionListener( new ActionListener()
    {
      public void actionPerformed( final ActionEvent event )
      {
        PushButton button = (PushButton) get();
        if ( 0 == select.getSelectedIndex() )
        {
          button.setEnabled( true );
        }
        else
        {
          button.setEnabled( false );
        }
      }
    });
    
    return select;
  }
}
